src/shims/mod.rs

```diff
@@ -27,9 +28,25 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         // There are some more lang items we want to hook that CTFE does not hook (yet).
         if this.tcx.lang_items().align_offset_fn() == Some(instance.def.def_id()) {
-            // FIXME: return a real value in case the target allocation has an
-            // alignment bigger than the one requested.
-            let n = u128::max_value();
+
+            let n = {
+                let ptr = this.read_scalar(args[0])?.not_undef()?.assert_ptr();
```

Comment:
If it reports a normal error that is fine. I just assumed `assert_ptr` to be reched and then panic instead of reporting an error.

Probably some earlier code bailing out

Reply:
Please add this as a test so we don't start ICEing at some point

Reply:
No, it's actually panicking. I changed this to `force_ptr` instead so it reports an error.